CuriousKatie/ViewController.swift

```diff
@@ -44,5 +44,39 @@ class ViewController: UIViewController {
                 print(somethingToShare)
             }
         }
+        
+        /// Print a new line to separate sections.
+        print("\n")
+        
+        /// Creating matching pairs of people
+        var seekingParticipants = participants
+        var participantsWithoutPartners = participants
+        var oldMatchingScore = 0
+        var matchingPartner = participants[0]
+        while seekingParticipants.count != 0 {
+            guard seekingParticipants.count != 1 else {
+                let lonelyParticipant = seekingParticipants[0]
+                print("\(lonelyParticipant.name) has no partner.")
+                seekingParticipants.removeAll { $0.name == lonelyParticipant.name }
+                break
+            }
+            matchingPartner = seekingParticipants[0]
+            for seeker in seekingParticipants {
+                for potentialPartner in participantsWithoutPartners {
+                    let matchingScore = Helper.calculateMatchingScore(between: seeker, and: potentialPartner)
```

Comment:
It looks it is possible for `seeker` and `potentialPartner` to be the same person